### PR TITLE
Updating curation document to include content lock ability

### PR DESCRIPTION
**Pull request by annezazu**
## What?

Building off of the launch of this new functionality in 6.1, this updates the curation resource to include information about content lock ability: https://github.com/WordPress/gutenberg/pull/43037 While there is currently documentation in place https://developer.wordpress.org/block-editor/reference-guides/block-api/block-templates/#locking this rounds out the current resources to provide practical examples, explain the benefits, and properly align it with other similar methods. 

@richtabor thanks to [your excellent article](https://richtabor.com/content-only-editing/) on the topic and your great feedback around naming, I'd love to have a review from you on this if you have time. I'm personally happy to fully roll with "content only editing" but wanted to keep it closer to what it was implemented as for now.

### CODE REVIEW

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (mburridge) [inline docs/how-to-guides/curating-the-editor-experience.md]: I had to read this several times in order to parse it.

Could it be rephrased as: "Child blocks of a locked block are also locked automatically and cannot be moved or removed"?

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (justintadlock) [inline docs/how-to-guides/curating-the-editor-experience.md]: Got a little tongue-twisted with this:

> In contrast to block locking, which removes the ability to remove or move blocks

Rolls off the tongue a little better:

> In contrast to block locking, which disables the ability to move or remove blocks

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (richtabor) [inline docs/how-to-guides/curating-the-editor-experience.md]: "Content locking" is the opposite of what's actually happening. We're locking down the structural/design attributes of the block, while keeping just the content editable. I think this will be quite confusing in the current wording.

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (richtabor) [inline docs/how-to-guides/curating-the-editor-experience.md]: Just confirming (and noting), but there's not a way to remove the "Modify" control right? It's always available, regardless of how locked the site is, yea?

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (richtabor) [inline docs/how-to-guides/curating-the-editor-experience.md]: To adopt this functionality, you need to manually add the `"templateLock":"contentOnly"` attribute to a block. 

Maybe mention that there's not a UI control for this — it's only possible via editing a pattern/part/template file. 

Also, we need to confirm what blocks this can be used on. Last I checked it may have only been group blocks that are supported. Doesn't work with Cover/Columns blocks I know.

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (richtabor) [inline docs/how-to-guides/curating-the-editor-experience.md:38]: Related to my earlier comment, this is not possible for contentOnly, correct?

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (annezazu) [reply docs/how-to-guides/curating-the-editor-experience.md]: Ah so in this case, it's more that child blocks within a container block that has this locking in place also are automatically locked. It's not the same as the block locking we're used to! I'll clarify the "children" part though to add a bit more context.

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (annezazu) [reply docs/how-to-guides/curating-the-editor-experience.md]: AFAIK yes (and based on my testing). @jorgefilipecosta can you confirm?  I see this was asked on the PR and went unanswered as well so I will follow up there too: https://github.com/WordPress/gutenberg/pull/43037#issuecomment-1263361207

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (annezazu) [reply docs/how-to-guides/curating-the-editor-experience.md]: Great call out around which blocks it can be added to. Seems like it is only Group. Can you confirm this too @jorgefilipecosta? ❤️

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (annezazu) [reply docs/how-to-guides/curating-the-editor-experience.md]: Agreed. This is exactly why I wanted your read of it :D Do you think we should roll with calling this "content only editing"?

Comment (richtabor) [reply docs/how-to-guides/curating-the-editor-experience.md]: Yea, I think that resonates (from feedback/responses of my article at least).

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (jorgefilipecosta) [reply docs/how-to-guides/curating-the-editor-experience.md]: Yes, right now, one can not remove modify button. The user can opt-out of this mode to temporarily make structure changes. In the future we may change this depending on role or capabilities.

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (jorgefilipecosta) [reply docs/how-to-guides/curating-the-editor-experience.md]: It can work on any block, even third-party blocks that have the templateLock attribute.
The templateLock attribute can be added to the block.json with
```
		"templateLock": {
			"type": [ "string", "boolean" ],
			"enum": [ "all", "insert", "contentOnly", false ]
		}
```

On core, it works with columns, covers, and groups.

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (jorgefilipecosta) [reply docs/how-to-guides/curating-the-editor-experience.md:38]: Yes for content lock this is not yet possible.

File: docs/how-to-guides/curating-the-editor-experience.md
Comment (annezazu) [reply docs/how-to-guides/curating-the-editor-experience.md]: Love it.

**Comment by annezazu** [comment]
Resolved all of the latest comments. Could I get additional reviews and, potentially, approvals? Of note, there's some discussion around implementing this on the dev note: https://make.wordpress.org/core/2022/10/11/content-locking-features-and-updates/

I might need to hold off on merging these doc updates depending on where things land.

**Comment by annezazu** [comment]
@ryanwelcher could you review this by chance? The feature is going to be in for 6.1 and I'd like to ensure the docs are ready.

**Comment by justintadlock** [comment]
Throwing another thought out there about "content-only" editing:

The phrasing is obviously weird because each of the other options refer to what they are **locking**.  However, this particular option refers to what it **allows**.

From a development perspective, I see this more as a "design locking."  What we're actually doing here is locking down the design tools.  

It's a bit late to change `"templateLock":"contentOnly"` to `"templateLock":"design"` at this point, but it might be helpful to refer to this as a design-locking feature in communications with developers.  I think it'd be easier to understand.

**Comment by annezazu** [comment]
Hmm. Thank you for the insights @justintadlock -- is this a blocker in your mind? I'm wondering if we can proceed as is considering the dev note and a "close enough" feeling. If you feel really strongly, happy to update though as I do not :) 

Curious what @richtabor thinks too.

**Comment by justintadlock** [comment]
> Hmm. Thank you for the insights @justintadlock -- is this a blocker in your mind? I'm wondering if we can proceed as is considering the dev note and a "close enough" feeling. If you feel really strongly, happy to update though as I do not :)

No strong feelings about it. Really, I was just making a note of this while the thought crossed my mind.

**Review by fabiankaegy — APPROVED** [review]
I also agree with the larger sentiment that the name still isn't ideal and I quite like the suggestion that @justintadlock posted. But I find this documentation update very valuable and should be merged now ahead of the 6.1 release to ensure we cover it in our documentation. 

We can always follow up with additional improvements going forward.

**Pull request merged**
